### PR TITLE
added remaining brew dependencies to aws

### PR DESCRIPTION
**Pull request by SAlexandru**
Added all brew dependencies from the provision script to AWS and create the osquery/osquery-local formulas for them 
Compiled this both on  ubuntu18.04 vagrant image and on mac os High Sierra

### CODE REVIEW

**Comment by theopolis** [comment]
Wow, this looks like a lot of work!

Quick question, should `make deps` work?

I see:
```
» make deps
/bin/bash: _platform: command not found
/bin/bash: _distro: command not found
./tools/provision.sh build 
/home/teddy/git/osquery/tools/provision/lib.sh: line 408: platform: command not found
Makefile:229: recipe for target 'deps' failed
make: *** [deps] Error 127

```

**Comment by SAlexandru** [comment]
@theopolis nothing in the build system should change, just from where the deps are downloaded. Don't know why, but somehow the wrong lib.sh got overwritten. Fixing this.

**Comment by guliashvili** [comment]
ok to test

File: tools/provision/formula/linux-headers.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    sha256 "e19f4291f8c1ef5c83d39f3a8f33b334ac83c686ff80bf71312a7b058c9c146a" => :x86_64_linux 
+  end
+
+  def install
+    system "make", "headers_install", "INSTALL_HDR_PATH=#{prefix}"
+    rm Dir[prefix/"**/{.install,..install.cmd}"]
+  end
+
+  test do
+    assert_match "KERNEL_VERSION", File.read(include/"linux/version.h")
```
[the remark this answers is not in the document]
Comment (SAlexandru) [reply tools/provision/formula/linux-headers.rb]: @akindyakov  yes it is different.  That include points to #{prefix}/include  (https://github.com/Homebrew/brew/blob/master/docs/Formula-Cookbook.md) where the file is actually copied :)

File: tools/provision/formula/clang-format.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      url "https://llvm.org/svn/llvm-project/llvm/tags/google/stable/2018-04-24/", :using => :svn
+    else
+      url "http://llvm.org/svn/llvm-project/llvm/tags/google/stable/2018-04-24/", :using => :svn
+    end
+
+    resource "clang" do
+      if MacOS.version >= :sierra
+        url "https://llvm.org/svn/llvm-project/cfe/tags/google/stable/2018-04-24/", :using => :svn
+      else
+        url "http://llvm.org/svn/llvm-project/cfe/tags/google/stable/2018-04-24/", :using => :svn
```
[the remark this answers is not in the document]
Comment (SAlexandru) [reply tools/provision/formula/clang-format.rb]: 👍

**Pull request merged**